composer/datasets/webdataset.py

```diff
@@ -123,7 +165,17 @@ def init_webdataset(remote: str,
 
 def size_webdataset(dataset: WebDataset, n_shards: int, samples_per_shard: int, n_devices: int, workers_per_device: int,
                     batch_size: int, drop_last: bool) -> WebDataset:
-    """Calculate WebDataset with_epoch() and with_length()."""
+    """Calculate WebDataset with_epoch() and with_length().
```

Comment:
There's a lot of detail in the math below that I think we need to describe in the docstrings here. e.g. dataset trunction, shards, etc.